app/src/main/java/com/devshish/internship/presentation/ui/lyrics/LyricsFragment.kt

```diff
@@ -19,16 +20,30 @@ class LyricsFragment : Fragment(R.layout.fragment_lyrics) {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
+        with(binding) {
+            ivSave.setOnClickListener {
+                viewModel.onLikeButtonClick(args.searchSong)
+            }
+        }
+
         with(viewModel) {
             getLyrics(args.searchSong)
```

Comment:
You can use koin to provide song from args into constructor of ViewModel, so you will not need to call this method from fragment